### ISSUE TITLE
There is multiple resolve for generic arguments if private and public fields are with the same name

### ISSUE THREAD

**Issue by mili-l** [intellij-rust/intellij-rust#8308]
## Environment

* **IntelliJ Rust plugin version:** 0.4.163.4319-213-nightly
* **Rust toolchain version:** 1.59.0-nightly (0b42deacc 2021-12-09) x86_64-apple-darwin
* **IDE name and version:** CLion 2021.3 (CL-213.5744.254)
* **Operating system:** macOS 11.5.1
* **Macro expansion engine:** new
* **Name resolution engine:** new
* **Additional experimental features:** org.rust.cargo.features.settings.gutter, org.rust.cargo.evaluate.build.scripts, org.rust.macros.proc

## Problem description
1) Multiple resolve for Bar
2) E0747 is not detected

## Steps to reproduce
```rust
pub mod foo {
    trait Bar {
    }
    pub struct Baz;

    pub fn Bar() { }
}

fn call_bar() {
    use foo::*;

    let _x: Box<Bar>;
}
```
## Actual Results
E0747 is not detected and multiple resolve for Bar

> error[E0747]: constant provided when a type was expected
>   --> src/main.rs:35:17
>    |
> 35 |     let _x: Box<Bar>;
>    |                 ^^^
>    |

<img width="488" alt="Screenshot 2021-12-20 at 11 07 26" src="https://user-images.githubusercontent.com/1213122/146757944-1a669630-4026-47f1-80a3-f74d78321c21.png">

Related to #8144
